applications/callflow/src/cf_exe.erl

```diff
@@ -655,6 +655,22 @@ code_change(_OldVsn, State, _Extra) ->
 %%% Internal functions
 %%%=============================================================================
 
+%%------------------------------------------------------------------------------
+%% @doc This function determines if the call is still active
+%% and launches the first action
+%% Otherwise it will exit.
+%% @end
+%%------------------------------------------------------------------------------
+-spec handle_callflow_ready(state()) ->  kz_types:handle_cast_ret_state(state()).
+handle_callflow_ready(#state{call=Call}=State) ->
+    CallId = kapps_call:call_id_direct(Call),
+    case kapps_call_command:b_channel_status(CallId) of
```

Comment:
I wonder if we hide this behind a system_config flag so we can optionally turn it on if needed? I really like the ecallmgr changes but this may be heavy-handed in the "normal" case.